### ISSUE TITLE
Windows unicode character entry doesn't work properly

### ISSUE THREAD

**Issue by davidearl** [WordPress/gutenberg#57981]
### Description

On Windows, if you hold down the ALT key and type one or more digits on the numeric keypad (not the main digits), it inserts an otherwise un-type-able character. For example, press ALT and then 2 and 6 on the numeric keypad, and release ALT inserts a right arrow, thus: →

In WordPress this does insert the character, but at the end of the document you are typing. So if you say add some headings and then start filling in content between them, the right arrow ends up at the end of the last headline rather than where the cursor is. It actually moves the insertion point, so if you carry on typing it all continues at the end of the document.

### Step-by-step reproduction instructions

* Start a new post
* add a couple of headings, say
* start a paragraph with a word or two after the first heading
* enter the Unicode character. e.g. 2 and 6 with the ALT key held down, then release it
* insertion point jumps to end of document before character appears, and ends up after that character.

### Screenshots, screen recording, code snippet

![image](https://github.com/WordPress/gutenberg/assets/1050651/e70bd126-24fe-4a9e-ae0f-830a7133a022)

### Environment info

WP 6.4.2 (Gutenberg whatever comes with that)
Theme: Sensible WP
Computer: Windows 10
Browser: Brave (screenshot), but checked in Chrome (120) too

I found #43352 but that seems more about missing shortcuts, whereas this is a bug

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by t-hamano** [comment]
Thanks for the report. This problem may be related not only to the Windows OS but also to the specifications and settings of the keyboard hardware.

For example, I typed `Alt + 0137` to type the character `‰`. Even if I press a number, the cursor position does not change and the character is inserted at the same position.

In your environment, does pressing a number move the cursor?

https://github.com/WordPress/gutenberg/assets/54422211/fdcf315b-aadd-4d08-a1d7-36a417ddcfbb

**Comment by davidearl** [comment]
Ah, right, I see what is happening...

It depends whether you have Num Lock on or not - that changes what keycode is sent. I did not but presumably you did. And with NumLock off the keystroke is 2 = down and 6 = right. Your choice of 0137 was, coincidentally, not on any of the arrow keys on the numeric keypad, though the ALT+0/INS key still does something peculiar when Num Lock is off, haven't quite worked out what exactly, but I get nothing at all.

So what's happening is Gutenberg is seeing those keys as arrow keys, even when the ALT key is depressed (and when released, Windows also does its standard unicode insertion). Presumably there is keydown event being picked up somewhere that is not checking for ALT. If I put Num Lock on it behaves as I would expect.

**Comment by t-hamano** [comment]
You're right, I have Num Lock enabled on my machine, and was able to reproduce this issue when I disabled it. I tried entering Unicode in the Windows Notes app or in a textarea on the web, but the cursor did not move.

Therefore, I think this is a problem that should be fixed on Gutenberg's side.

**Comment by ellatrix** [comment]
Unfortunately, I don't have a windows device, so it's really hard for me to work on a fix. Would appreciate if someone with Windows could give it a try.
